packages/block-library/src/columns/edit.native.js

```diff
@@ -0,0 +1,279 @@
+ * In columns block, the only block we allow is 'core/column'.
+ *
+ * @constant
+ * @type {string[]}
+ */
+const ALLOWED_BLOCKS = [ 'core/column' ];
+
+/**
+ * Number of columns to assume for template in case the user opts to skip
+ * template option selection.
+ *
+ * @type {number}
+ */
+const DEFAULT_COLUMNS = 2;
+const MIN_COLUMNS_NUMBER = 1;
+
+const BREAKPOINTS = {
+	mobile: 480,
+	large: 768,
+};
+
+function ColumnsEditContainer( {
+	attributes,
+	updateAlignment,
+	updateColumns,
+	columnCount,
+	isSelected,
+	onAddNextColumn,
+	onDelete,
+} ) {
+	const [ resizeListener, sizes ] = useResizeObserver();
+	const [ columnsSettings, setColumnsSettings ] = useState( {
+		columnsInRow: MIN_COLUMNS_NUMBER,
+	} );
+
+	const { verticalAlignment } = attributes;
+	const { width } = sizes || {};
+
+	useEffect( () => {
+		const newColumnCount = ! columnCount ? DEFAULT_COLUMNS : columnCount;
+		updateColumns( columnCount, newColumnCount );
+		setColumnsSettings( {
+			...columnsSettings,
+			columnsInRow: getColumnsInRow( width, newColumnCount ),
+		} );
+	}, [ columnCount ] );
+
+	useEffect( () => {
+		setColumnsSettings( {
+			columnsInRow: getColumnsInRow( width, columnCount ),
+			width,
+		} );
+	}, [ width ] );
+
+	const getColumnsInRow = ( containerWidth, columnsNumber ) => {
+		if ( containerWidth < BREAKPOINTS.mobile ) {
+			// show only 1 Column in row for mobile breakpoint container width
+			return 1;
+		} else if ( containerWidth < BREAKPOINTS.large ) {
+			// show 2 Column in row for large breakpoint container width
+			return Math.min( columnCount, 2 );
+		}
+		// show all Column in one row
+		return columnsNumber;
+	};
+
+	const renderAppender = () => {
+		if ( isSelected ) {
+			return (
+				<InnerBlocks.ButtonBlockAppender
+					customOnAdd={ onAddNextColumn }
+				/>
+			);
+		}
+		return null;
+	};
+
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Columns Settings' ) }>
+					<StepperControl
+						label={ __( 'Number of columns' ) }
+						icon="columns"
+						value={ columnCount }
+						onChange={ ( value ) =>
+							updateColumns( columnCount, value )
+						}
+						min={ MIN_COLUMNS_NUMBER }
+						max={ columnCount + 1 }
+						separatorType={ 'none' }
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<BlockControls>
+				<BlockVerticalAlignmentToolbar
+					onChange={ updateAlignment }
+					value={ verticalAlignment }
+				/>
+			</BlockControls>
+			<View style={ isSelected && styles.innerBlocksSelected }>
+				{ resizeListener }
+				<InnerBlocks
+					renderAppender={ renderAppender }
+					__experimentalMoverDirection={
+						getColumnsInRow( width, columnCount ) > 1
+							? 'horizontal'
+							: undefined
+					}
+					flatListProps={ {
+						contentContainerStyle: styles.columnsContainer,
+						horizontal: true,
+						scrollEnabled: false,
+						style: styles.innerBlocks,
+					} }
+					allowedBlocks={ ALLOWED_BLOCKS }
+					customBlockProps={ {
+						...columnsSettings,
```

Comment:
Can't we calculate the "columnWidth" inside columns/edit.native.js instead of column/edit.native.js and pass here only the result width?

We can then pass it inside a kind of style prop(we can use a different name like "contentStyle" if that makes more sense) ?

Reply:
I think it's doable will work on that

Reply:
done